src/course-unit/xblock-container-iframe/index.tsx

```diff
@@ -1,57 +1,183 @@
-import { useRef, useEffect, FC } from 'react';
-import PropTypes from 'prop-types';
+import {
+  useRef, FC, useEffect, useState, useMemo, useCallback,
+} from 'react';
 import { useIntl } from '@edx/frontend-platform/i18n';
 import { getConfig } from '@edx/frontend-platform';
+import { useToggle } from '@openedx/paragon';
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
 
-import { IFRAME_FEATURE_POLICY } from '../constants';
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import ConfigureModal from '../../generic/configure-modal/ConfigureModal';
+import { copyToClipboard } from '../../generic/data/thunks';
+import { COURSE_BLOCK_NAMES, IFRAME_FEATURE_POLICY } from '../../constants';
+import { messageTypes, COMPONENT_TYPES } from '../constants';
+import { fetchCourseUnitQuery } from '../data/thunk';
 import { useIframe } from '../context/hooks';
 import { useIFrameBehavior } from './hooks';
 import messages from './messages';
 
-/**
- * This offset is necessary to fully display the dropdown actions of the XBlock
- * in case the XBlock does not have content inside.
- */
-const IFRAME_BOTTOM_OFFSET = 220;
+import {
+  XBlockContainerIframeProps,
+  XBlockDataTypes,
+  MessagePayloadTypes,
+} from './types';
 
-interface XBlockContainerIframeProps {
-  blockId: string;
-}
-
-const XBlockContainerIframe: FC<XBlockContainerIframeProps> = ({ blockId }) => {
+const XBlockContainerIframe: FC<XBlockContainerIframeProps> = ({
+  courseId, blockId, unitXBlockActions, courseVerticalChildren, handleConfigureSubmit,
+}) => {
   const intl = useIntl();
   const iframeRef = useRef<HTMLIFrameElement>(null);
-  const { setIframeRef } = useIframe();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const [isConfigureModalOpen, openConfigureModal, closeConfigureModal] = useToggle(false);
+  const { setIframeRef, sendMessageToIframe } = useIframe();
+  const [currentXBlockId, setCurrentXBlockId] = useState<string | null>(null);
+  const [currentXBlockData, setCurrentXBlockData] = useState<any>({});
```

Comment:
Please specify an actual type, not `any`.

What is the "current XBlock" ?

Reply:
[inform] Implemented in follow-up commit: https://github.com/openedx/frontend-app-authoring/pull/1431/commits/ea4da9f7625c3ca176d5fd8132f0b23a8329886c#diff-b3b994409f1cdaa22d4d3c9cdba42318405a2bcaf82c1f91056eca6443f7448aR39